internal/controller/pipelinerollout_controller.go

```diff
@@ -168,7 +168,7 @@ func (r *PipelineRolloutReconciler) reconcile(
 		}
 
 		kubernetes.SetAnnotation(pipelineRollout, apiv1.KeyHash, childResouceSpecHash)
-	} else {
+	} else if rolloutChildOp != RolloutChildNoop {
```

Comment:
nit: I'd prefer this to be `if rolloutChildOp == RolloutChildUpdate` to be more clear.

Reply:
yeah, I was thinking the same. The logic below applies to when we're trying to update.

Reply:
I actually just realized that we need to remove this. I had it in place in the previous PR but it needs to be removed because part of the logic inside the functions called in this if-statement branch need to execute in case the pipeline needs to be updated or not. The code in this branch needs to be refactored, but I wanted to do that in a separate PR as discussed with @xdevxy

Reply:
> I actually just realized that we need to remove this. I had it in place in the previous PR but it needs to be removed because part of the logic inside the functions called in this if-statement branch need to execute in case the pipeline needs to be updated or not.

You know, I made that comment back when you were updating the annotation immediately whenever there was a need to update. Now that you're not updating that annotation until the update has truly happened, I think it may be okay. Basically, we will revisit this "update" logic up until the update actually occurs.

Reply:
> The code in this branch needs to be refactored

are you referring partly to my other comment regarding the long function name here?